continuous-integration/check.sh

```diff
@@ -0,0 +1,248 @@
+#!/bin/bash -e
+
+function do_Install_Base_Dependencies(){
+    echo
+    echo '-- Installing Base Dependencies --'
+
+    if [[ $BASE == "ubuntu" ]]; then
+        apt-get update -qq
+
+        # Base dependencies
+        apt-get -y -qq install build-essential git clang patch > /dev/null
```

Comment:
Could we stick all these in one big `apt-get -y -qq install` command? It would probably cut down some time on resolving dependencies.